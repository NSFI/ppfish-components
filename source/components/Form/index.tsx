import Form from './Form';
import './style/index.less';

export {
  FormProps,
  FormComponentProps,
  FormCreateOption,
  ValidateCallback,
  ValidationRule
} from './Form';
export { FormListProps } from './FormList';
export { FormItemProps } from './FormItem';

export default Form;
