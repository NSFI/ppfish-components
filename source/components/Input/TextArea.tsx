import * as React from 'react';
import omit from 'omit.js';
import classNames from 'classnames';
import calculateNodeHeight from './calculateNodeHeight';

function onNextFrame(cb: () => void) {
  if (window.requestAnimationFrame) {
    return window.requestAnimationFrame(cb);
  }
  return window.setTimeout(cb, 1);
}

function clearNextFrameAction(nextFrameId: number) {
  if (window.cancelAnimationFrame) {
    window.cancelAnimationFrame(nextFrameId);
  } else {
    window.clearTimeout(nextFrameId);
  }
}

export interface AutoSizeType {
  minRows?: number;
  maxRows?: number;
}

export type HTMLTextareaProps = React.TextareaHTMLAttributes<HTMLTextAreaElement>;

export interface TextAreaProps extends HTMLTextareaProps {
  prefixCls?: string;
  autosize?: boolean | AutoSizeType;
  onPressEnter?: React.KeyboardEventHandler<HTMLTextAreaElement>;
}

export interface TextAreaRef {
  textAreaRef: HTMLTextAreaElement;
  focus: () => void;
  blur: () => void;
  resizeTextarea: () => void;
}

const InternalTextArea: React.ForwardRefRenderFunction<TextAreaRef, TextAreaProps> = (
  props,
  ref,
) => {
  const [textareaStyles, setTextareaStyles] = React.useState<React.CSSProperties>({});
  const nextFrameActionIdRef = React.useRef<number>(null);
  const textAreaRef = React.useRef<HTMLTextAreaElement>(null);

  const resizeTextarea = () => {
    const { autosize } = props;
    const textAreaNode = textAreaRef.current;
    if (!autosize || !textAreaNode) {
      return;
    }
    const minRows = autosize ? (autosize as AutoSizeType).minRows : null;
    const maxRows = autosize ? (autosize as AutoSizeType).maxRows : null;
    const textareaStyles = calculateNodeHeight(textAreaNode, false, minRows, maxRows);
    setTextareaStyles(textareaStyles);
  };

  const getTextAreaClassName = () => {
    const { prefixCls, className, disabled } = props;
    return classNames(prefixCls, className, {
      [`${prefixCls}-disabled`]: disabled,
    });
  };

  const handleTextareaChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    if (!('value' in props)) {
      resizeTextarea();
    }
    const { onChange } = props;
    if (onChange) {
      onChange(e);
    }
  };

  const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
    const { onPressEnter, onKeyDown } = props;
    if (e.keyCode === 13 && onPressEnter) {
      onPressEnter(e);
    }
    if (onKeyDown) {
      onKeyDown(e);
    }
  };

  React.useImperativeHandle(ref, () => ({
    textAreaRef: textAreaRef.current,
    focus: () => {
      textAreaRef.current.focus();
    },
    blur: () => {
      textAreaRef.current.blur();
    },
    resizeTextarea,
  }));

  React.useEffect(() => {
    const nextFrameActionId = nextFrameActionIdRef.current;
    if (nextFrameActionId) {
      clearNextFrameAction(nextFrameActionId);
    }
    nextFrameActionIdRef.current = onNextFrame(resizeTextarea);
  }, [props.value]);

  const otherProps = omit(props, ['prefixCls', 'onPressEnter', 'autosize']);
  const style = {
    ...props.style,
    ...textareaStyles,
  };
  // Fix https://github.com/ant-design/ant-design/issues/6776
  // Make sure it could be reset when using form.getFieldDecorator
  if ('value' in otherProps) {
    otherProps.value = otherProps.value || '';
  }
  return (
    <textarea
      {...otherProps}
      className={getTextAreaClassName()}
      style={style}
      onKeyDown={handleKeyDown}
      onChange={handleTextareaChange}
      ref={textAreaRef}
    />
  );
};

const TextArea = React.forwardRef(InternalTextArea);

TextArea.displayName = 'TextArea';

TextArea.defaultProps = {
  prefixCls: 'fishd-input',
};

export default TextArea;
