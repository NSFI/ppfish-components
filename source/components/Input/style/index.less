@import "../../../assets/css/index";
@import "./mixin";

// Input styles
.@{fishd-prefix}-input {
  .reset-component;
  .input;
}

//== Style for input-group: input with label, with button or dropdown...
.@{fishd-prefix}-input-group {
  .reset-component;
  .input-group(~"@{fishd-prefix}-input");
  &-wrapper {
    display: inline-block;
    vertical-align: top; // https://github.com/ant-design/ant-design/issues/6403
    width: 100%;
  }
}

// Input with affix: prefix or suffix
.@{fishd-prefix}-input-affix-wrapper {
  .reset-component;
  .input-affix-wrapper(~"@{fishd-prefix}-input");

  // https://github.com/ant-design/ant-design/issues/6144
  .@{fishd-prefix}-input {
    min-height: 100%;  // use min-height, assume that no smaller height to override
  }
}

@import "./search-input";
