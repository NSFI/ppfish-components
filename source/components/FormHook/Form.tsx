import  React, { useMemo } from 'react';
import classNames from 'classnames';
import FieldForm, { List } from './rcFieldForm';
import { FormProps as RcFormProps } from './rcFieldForm/Form';
import { ValidateErrorEntity } from './rcFieldForm/interface';
import { Options } from 'scroll-into-view-if-needed';
import { ColProps } from '../Grid';
import { FormContext, FormContextProps } from './context';
import { FormLabelAlign } from './interface';
import useForm, { FormInstance } from './hooks/useForm';
export type RequiredMark = boolean | 'optional';
export type FormLayout = 'horizontal' | 'inline' | 'vertical';

/**
 * 没有添加 Size  需要有一个全局的 SizeContext, 所有组件都要配合
 * config context 中的全局配置未添加 getPrefixCls, direction, form 等情况
 * RequiredMark - optional, rtl 暂不实现
 * colon 特意删除
 *
 * antd 中 form 的表单标签可换行功能 需要 Col 组件的支持
 *
 * 缺少 Space 组件
 *
 * 因为 Col 组件的不一样, 导致 form 的 layout 即使默认是horizontal, 如果没传span, 也会和 vertical 一样
 *
 * FormItem hasFeedback 属性没有 FormItemStatusContext 支持, 这里使用老的一套 有.has-feedback属性 input 则显示
 *
 * 置空存在在现在的Col 组件上存在动画问题, 查看高级搜索 demo, 去除 fields 样式
 *
 * DatePicker 中不能选择月, 待加强
 *
 * DateRangePicker 第一次点击时有 warning 需要处理  `DateTable: prop type `rangeState.endDate` is invalid`
 */

export interface FormProps<Values = any> extends Omit<RcFormProps<Values>, 'form'> {
  prefixCls?: string;
  name?: string;
  layout?: FormLayout;
  labelAlign?: FormLabelAlign;
  labelWrap?: boolean;
  labelCol?: ColProps;
  wrapperCol?: ColProps;
  form?: FormInstance<Values>;
  scrollToFirstError?: Options | boolean;
  hideRequiredMark?: boolean;
}

const InternalForm: React.ForwardRefRenderFunction<FormInstance, FormProps> = (props, ref) => {

  const {
    prefixCls,
    className = '',
    form,
    labelAlign,
    labelWrap,
    labelCol,
    wrapperCol,
    hideRequiredMark,
    layout = 'horizontal',
    scrollToFirstError,
    onFinishFailed,
    name,
    ...restFormProps
  } = props;

  const formClassName = classNames(
    prefixCls,
    {
      [`${prefixCls}-horizontal`]: layout === 'horizontal',
      [`${prefixCls}-vertical`]: layout === 'vertical',
      [`${prefixCls}-inline`]: layout === 'inline',
      [`${prefixCls}-hide-required-mark`]: hideRequiredMark,
    },
    className,
  );

  const [wrapForm] = useForm(form);
  const { __INTERNAL__ } = wrapForm;
  __INTERNAL__.name = name;

  const formContextValue = useMemo<FormContextProps>(
    () => ({
      name,
      labelAlign,
      labelCol,
      labelWrap,
      wrapperCol,
      vertical: layout === 'vertical',
      requiredMark: hideRequiredMark,
      itemRef: __INTERNAL__.itemRef,
    }),
    [name, labelAlign, labelCol, wrapperCol, layout, hideRequiredMark],
  );

  React.useImperativeHandle(ref, () => wrapForm);

  const onInternalFinishFailed = (errorInfo: ValidateErrorEntity) => {
    onFinishFailed?.(errorInfo);

    let defaultScrollToFirstError: Options = { block: 'nearest' };

    if (scrollToFirstError && errorInfo.errorFields.length) {
      if (typeof scrollToFirstError === 'object') {
        defaultScrollToFirstError = scrollToFirstError;
      }
      wrapForm.scrollToField(errorInfo.errorFields[0].name, defaultScrollToFirstError);
    }
  };

  return (
      <FormContext.Provider value={formContextValue}>
        <FieldForm
          id={name}
          {...restFormProps}
          name={name}
          onFinishFailed={onInternalFinishFailed}
          form={wrapForm}
          className={formClassName}
        />
      </FormContext.Provider>
  );
};




const Form = React.forwardRef<FormInstance, FormProps>(InternalForm) as <Values = any>(
  props: React.PropsWithChildren<FormProps<Values>> & { ref?: React.Ref<FormInstance<Values>> },
) => React.ReactElement;

(Form as any).defaultProps = {
  prefixCls: 'fishd-formHook',
  layout: 'horizontal',
  labelAlign: 'left', // 按照 Form 的标准 默认靠左
  hideRequiredMark: false,
  onSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();
  },
}


export { useForm, List, FormInstance };

export default Form;
