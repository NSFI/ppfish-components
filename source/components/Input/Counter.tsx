import * as React from 'react';
import omit from 'omit.js';
import classNames from 'classnames';
import TextArea, { TextAreaProps, TextAreaRef } from './TextArea';
import useControlledState from '../../hooks/useControlledState';

function countValue(value: string) {
  return value.length;
}

export interface CounterProps extends TextAreaProps {
  inputPrefixCls?: string;
  prefixCls?: string;
  limit: number;
  count?: (value: string) => number;
  value?: any;
  defaultValue?: any;
}

export interface CounterRef {
  focus: () => void;
  blur: () => void;
  textarea: TextAreaRef;
}

const InternalCounter: React.ForwardRefRenderFunction<CounterRef, CounterProps> = (props, ref) => {
  const [value, setValue] = useControlledState('', {
    value: props.value,
    defaultValue: props.defaultValue,
  });
  const textareaRef = React.useRef<TextAreaRef>(null);

  const focus = () => {
    textareaRef.current.focus();
  };

  const handleClick = () => {
    focus();
  };

  const handleTextareaChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    const { onChange } = props;
    const textarea = textareaRef.current;
    const textareaValue = textarea && textarea.textAreaRef.value;
    setValue(textareaValue);
    if (onChange) {
      onChange(e);
    }
  };

  const getCount = () => {
    const { count } = props;
    if (!value) {
      return 0;
    }
    // 自定义计数方法
    if (count) {
      return count(String(value));
    }
    return countValue(String(value));
  };

  React.useImperativeHandle(ref, () => ({
    focus,
    blur: () => {
      textareaRef.current.blur();
    },
    textarea: textareaRef.current,
  }));

  const { inputPrefixCls, className, prefixCls, disabled, limit } = props;
  const inputClassName = classNames(className, {
    [`${prefixCls}`]: true,
    [`${inputPrefixCls}-disabled`]: disabled,
  });
  const textareaClassName = classNames(inputPrefixCls, className);
  const otherProps = omit(props, [
    'inputPrefixCls',
    'prefixCls',
    'limit',
    'count',
    'value',
    'onChange',
  ]);
  const total = getCount();
  return (
    <span className={inputClassName} onClick={handleClick}>
      <TextArea
        {...otherProps}
        className={textareaClassName}
        maxLength={limit}
        onChange={handleTextareaChange}
        value={value}
        ref={textareaRef}
      />
      <span className={`${prefixCls}-footer`}>
        <span className={`${prefixCls}-indicator`}>
          {total}/{limit}
        </span>
      </span>
    </span>
  );
};

const Counter = React.forwardRef(InternalCounter);

Counter.displayName = 'Counter';

Counter.defaultProps = {
  inputPrefixCls: 'fishd-input',
  prefixCls: 'fishd-input-counter',
};

export default Counter;
