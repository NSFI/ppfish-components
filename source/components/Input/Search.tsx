import * as React from 'react';
import * as classNames from 'classnames';
import Input, { InputProps } from './Input';
import Icon from '../Icon';
import Button from '../Button';

export interface SearchProps extends InputProps {
  inputPrefixCls?: string;
  onSearch?: (value: string, event?: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLInputElement>) => any;
  enterButton?: boolean | React.ReactNode;
}

export default class Search extends React.Component<SearchProps, any> {
  static defaultProps = {
    inputPrefixCls: 'fishd-input',
    prefixCls: 'fishd-input-search',
    enterButton: false,
  };

  private input: Input;

  onSearch = (e: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLInputElement>) => {
    const { onSearch } = this.props;
    if (onSearch) {
      onSearch(this.input.input.value, e);
    }
    this.input.focus();
  }

  focus() {
    this.input.focus();
  }

  blur() {
    this.input.blur();
  }

  saveInput = (node: Input) => {
    this.input = node;
  }

  getButtonOrIcon() {
    const { enterButton, prefixCls, size, disabled } = this.props;
    const enterButtonAsElement = enterButton as React.ReactElement<any>;
    let node;
    if (!enterButton) {
      node = <Icon className={`${prefixCls}-icon`} type="search" key="searchIcon" />;
    } else if (enterButtonAsElement.type === Button || enterButtonAsElement.type === 'button') {
      node = React.cloneElement(enterButtonAsElement, enterButtonAsElement.type === Button ? {
        className: `${prefixCls}-button`,
        size,
      } : {});
    } else {
      node = (
        <Button
          className={`${prefixCls}-button`}
          type="primary"
          size={size}
          disabled={disabled}
          key="enterButton"
        >
          {enterButton === true ? <Icon type="search" /> : enterButton}
        </Button>
      );
    }
    return React.cloneElement(node, {
      onClick: this.onSearch,
    });
  }

  render() {
    const { className, prefixCls, inputPrefixCls, size, suffix,  enterButton, ...others } = this.props;
    delete (others as any).onSearch;
    const buttonOrIcon = this.getButtonOrIcon();
    const searchSuffix = suffix ? [suffix, buttonOrIcon] : buttonOrIcon;
    const inputClassName = classNames(prefixCls, className, {
      [`${prefixCls}-enter-button`]: !!enterButton,
      [`${prefixCls}-${size}`]: !!size,
    });
    return (
      <Input
        onPressEnter={this.onSearch}
        {...others}
        size={size}
        className={inputClassName}
        prefixCls={inputPrefixCls}
        suffix={searchSuffix}
        ref={this.saveInput}
      />
    );
  }
}
