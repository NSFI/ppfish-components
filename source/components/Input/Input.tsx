import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import omit from 'omit.js';
import Group from './Group';
import Search from './Search';
import TextArea from './TextArea';
import Counter from './Counter';
import { Omit } from '../../utils/type';

function fixControlledValue<T>(value: T) {
  if (typeof value === 'undefined' || value === null) {
    return '';
  }
  return value;
}

export interface InputProps
  extends Omit<React.InputHTMLAttributes<HTMLInputElement>, 'size' | 'prefix'> {
  prefixCls?: string;
  className?: string;
  size?: 'large' | 'default' | 'small';
  onPressEnter?: React.KeyboardEventHandler<HTMLInputElement>;
  addonBefore?: React.ReactNode;
  addonAfter?: React.ReactNode;
  prefix?: React.ReactNode;
  suffix?: React.ReactNode;
}

export interface InputRef {
  input: HTMLInputElement;
  focus: () => void;
  blur: () => void;
}

const InternalInput: React.ForwardRefRenderFunction<InputRef, InputProps> = (props, ref) => {
  const inputRef = React.useRef(null);
  React.useImperativeHandle(ref, () => ({
    input: inputRef.current,
    focus: () => {
      inputRef.current.focus();
    },
    blur: () => {
      inputRef.current.blur();
    },
  }));

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    const { onPressEnter, onKeyDown } = props;
    if (e.keyCode === 13 && onPressEnter) {
      onPressEnter(e);
    }
    if (onKeyDown) {
      onKeyDown(e);
    }
  };

  const getInputClassName = () => {
    const { prefixCls, size, disabled } = props;
    return classNames(prefixCls, {
      [`${prefixCls}-sm`]: size === 'small',
      [`${prefixCls}-lg`]: size === 'large',
      [`${prefixCls}-disabled`]: disabled,
    });
  };

  const renderLabeledInput = (children: React.ReactElement<any>) => {
    // Not wrap when there is not addons
    if (!props.addonBefore && !props.addonAfter) {
      return children;
    }

    const wrapperClassName = `${props.prefixCls}-group`;
    const addonClassName = `${wrapperClassName}-addon`;
    const addonBefore = props.addonBefore ? (
      <span className={addonClassName}>{props.addonBefore}</span>
    ) : null;

    const addonAfter = props.addonAfter ? (
      <span className={addonClassName}>{props.addonAfter}</span>
    ) : null;

    const className = classNames(`${props.prefixCls}-wrapper`, {
      [wrapperClassName]: addonBefore || addonAfter,
    });

    const groupClassName = classNames(`${props.prefixCls}-group-wrapper`, {
      [`${props.prefixCls}-group-wrapper-sm`]: props.size === 'small',
      [`${props.prefixCls}-group-wrapper-lg`]: props.size === 'large',
    });

    // Need another wrapper for changing display:table to display:inline-block
    // and put style prop in wrapper
    if (addonBefore || addonAfter) {
      return (
        <span className={groupClassName} style={props.style}>
          <span className={className}>
            {addonBefore}
            {React.cloneElement(children, { style: null })}
            {addonAfter}
          </span>
        </span>
      );
    }
    return (
      <span className={className}>
        {addonBefore}
        {children}
        {addonAfter}
      </span>
    );
  };

  const renderLabeledIcon = (children: React.ReactElement<any>) => {
    if (!('prefix' in props || 'suffix' in props)) {
      return children;
    }

    const prefix = props.prefix ? (
      <span className={`${props.prefixCls}-prefix`}>{props.prefix}</span>
    ) : null;

    const suffix = props.suffix ? (
      <span className={`${props.prefixCls}-suffix`}>{props.suffix}</span>
    ) : null;

    const affixWrapperCls = classNames(props.className, `${props.prefixCls}-affix-wrapper`, {
      [`${props.prefixCls}-affix-wrapper-sm`]: props.size === 'small',
      [`${props.prefixCls}-affix-wrapper-lg`]: props.size === 'large',
    });
    return (
      <span className={affixWrapperCls} style={props.style}>
        {prefix}
        {React.cloneElement(children, {
          style: null,
          className: getInputClassName(),
        })}
        {suffix}
      </span>
    );
  };

  const renderInput = () => {
    const { value, className } = props;
    // Fix https://fb.me/react-unknown-prop
    const otherProps = omit(props, [
      'prefixCls',
      'onPressEnter',
      'addonBefore',
      'addonAfter',
      'prefix',
      'suffix',
    ]);

    if ('value' in props) {
      otherProps.value = fixControlledValue(value);
      // Input elements must be either controlled or uncontrolled,
      // specify either the value prop, or the defaultValue prop, but not both.
      delete otherProps.defaultValue;
    }
    return renderLabeledIcon(
      <input
        {...otherProps}
        className={classNames(getInputClassName(), className)}
        onKeyDown={handleKeyDown}
        ref={inputRef}
      />,
    );
  };

  return renderLabeledInput(renderInput());
};

interface InputInterface
  extends React.ForwardRefExoticComponent<InputProps & React.RefAttributes<InputRef>> {
  Group: typeof Group;
  Search: typeof Search;
  TextArea: typeof TextArea;
  Counter: typeof Counter;
}

const Input = React.forwardRef(InternalInput) as InputInterface;

Input.displayName = 'Input';

Input.defaultProps = {
  prefixCls: 'fishd-input',
  type: 'text',
  disabled: false,
};

Input.propTypes = {
  type: PropTypes.string,
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  size: PropTypes.oneOf(['small', 'default', 'large']),
  maxLength: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  disabled: PropTypes.bool,
  value: PropTypes.any,
  defaultValue: PropTypes.any,
  className: PropTypes.string,
  addonBefore: PropTypes.node,
  addonAfter: PropTypes.node,
  prefixCls: PropTypes.string,
  onPressEnter: PropTypes.func,
  onKeyDown: PropTypes.func,
  onKeyUp: PropTypes.func,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  prefix: PropTypes.node,
  suffix: PropTypes.node,
};

export default Input;
