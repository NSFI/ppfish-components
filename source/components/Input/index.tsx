import Input from './Input';
import Group from './Group';
import Search from './Search';
import TextArea from './TextArea';
import './style/index.less';

export {InputProps} from './Input';
export {GroupProps} from './Group';
export {SearchProps} from './Search';
export {TextAreaProps} from './TextArea';

Input.Group = Group;
Input.Search = Search;
Input.TextArea = TextArea;
export default Input;
