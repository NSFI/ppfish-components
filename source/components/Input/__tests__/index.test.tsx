import React from 'react';
import { mount } from 'enzyme';
import Input from '../index';
import Form from '../../Form';

describe('Input', () => {
  it('should support maxLength', () => {
    const wrapper = mount(<Input maxLength={3} />);
    const input = wrapper.at(0).find('.fishd-input').render();
    expect(input.attr('maxlength')).toBe('3');
  });

  it('should support focus() blur(), and can get input element', function () {
    const ref = React.createRef<any>();
    mount(<Input ref={ref} />);
    ref.current.focus();
    ref.current.blur();
    expect(ref.current.input instanceof HTMLInputElement).toBe(true);
  });

  it('should support size', () => {
    const wrapper = mount(<Input size="large" />);
    expect(wrapper.find('input').hasClass('fishd-input-lg')).toBe(true);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should support className when has suffix', () => {
    const wrapper = mount(<Input suffix="suffix" className="my-class-name" />);
    expect(wrapper.getDOMNode().className.includes('my-class-name')).toBe(true);
    expect(wrapper.find('input').getDOMNode().className.includes('my-class-name')).toBe(false);
  });

  it('should support className when has prefix', () => {
    const wrapper = mount(<Input prefix="prefix" className="my-class-name" />);
    expect(wrapper.getDOMNode().className.includes('my-class-name')).toBe(true);
    expect(wrapper.find('input').getDOMNode().className.includes('my-class-name')).toBe(false);
  });

  it('should be reset when wrapped in form.getFieldDecorator without initialValue', () => {
    const Demo = props => {
      const { form } = props;
      const { getFieldDecorator } = form;
      const reset = () => {
        form.resetFields();
      };
      return (
        <Form>
          <Form.Item>{getFieldDecorator('input')(<Input />)}</Form.Item>
          <Form.Item>{getFieldDecorator('textarea')(<Input.TextArea />)}</Form.Item>
          <button type="button" onClick={reset}>
            reset
          </button>
        </Form>
      );
    };
    const WrapDemo = Form.create()(Demo);

    const wrapper = mount(<WrapDemo />);
    wrapper.find('input').simulate('change', { target: { value: '111' } });
    wrapper.find('textarea').simulate('change', { target: { value: '222' } });
    expect(wrapper.find('input').prop('value')).toBe('111');
    expect(wrapper.find('textarea').prop('value')).toBe('222');
    wrapper.find('button').simulate('click');
    expect(wrapper.find('input').prop('value')).toBe('');
    expect(wrapper.find('textarea').prop('value')).toBe('');
  });
});
