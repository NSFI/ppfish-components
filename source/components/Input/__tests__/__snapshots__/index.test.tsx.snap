// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should support size 1`] = `
<input
  class="fishd-input fishd-input-lg"
  type="text"
/>
`;
